src/compiler/gcc.rs

```diff
@@ -349,10 +348,6 @@ pub fn preprocess<T>(creator: &T,
     let mut cmd = creator.clone().new_command_sync(executable);
     cmd.arg("-x").arg(language)
         .arg("-E");
-    // With -fprofile-generate line number information is important, so don't use -P.
-    if !parsed_args.profile_generate {
-        cmd.arg("-P");
-    }
```

Comment:
With distributed compilation, line number information is always important. More generally, it's important if you want to use a preprocessed file as input.

Reply:
Hm. This will regress the change in https://github.com/mozilla/sccache/pull/208. Would it be possible to change things so that when distributed compilation is not enabled we can continue to use this?

Reply:
Oh, this is disappointing, I should've done a bit more digging.

I'm not sure if it's better to say "distributed compilation *may* happen, your cache is completely invalid", or just preprocess a second time on-demand when sending for remote compilation. Looking at a reasonably large file of C++ code I have easily to hand, it's spending 10-20% of wall-clock time in preprocessing so running twice isn't ideal.

What about: if we *might* be distributing, do the full preprocess with line numbers but strip out the line numbers with a regexp when considering the cache key.

Reply:
That sounds fine. Since users have to explicitly opt-in to using distributed compilation, I don't think it's unreasonable for them to get this behavior in that case. Running the preprocessor twice seems like too much overhead just for this purpose. We can make a note somewhere if necessary that cache hits using different source paths are not possible with distributed compilation enabled.